src/equations/ideal_mhd_multiion_2d.jl

```diff
@@ -139,12 +139,12 @@ end
 
   mag_en = 0.5*(B1^2 + B2^2 + B3^2)
 
-  if orientation == 1
-    f_B1 = 0.0
-    f_B2 = v1_plus * B2 - v2_plus * B1
-    f_B3 = v1_plus * B3 - v3_plus * B1
+  f = zeros(MVector{nvariables(equations), eltype(u)})
```

Comment:
Does this really create an MVector? Why do you not use one of the standard constructors? At first, it looks like you create an array filled with zero MVectors

Reply:
I took this from https://juliaarrays.github.io/StaticArrays.jl/stable/pages/quickstart/, see `v5`. If you think a different style would be more legible, I’m all ears!

Reply:
I'm asking since it does not appear to have a docstring and does not coincide with the base docstring of `zeros`:
```julia
julia> using StaticArrays

help?> zeros
search: zeros count_zeros set_zero_subnormals get_zero_subnormals leading_zeros trailing_zeros zero iszero

  zeros([T=Float64,] dims::Tuple)
  zeros([T=Float64,] dims...)

  Create an Array, with element type T, of all zeros with size specified by dims. See also fill, ones, zero.

  Examples
  ≡≡≡≡≡≡≡≡≡≡

  julia> zeros(1)
  1-element Vector{Float64}:
   0.0
  
  julia> zeros(Int8, 2, 3)
  2×3 Matrix{Int8}:
   0  0  0
   0  0  0
```
Based on this docstring, the constructor looks suspiciously close to
```julia
julia> zeros(MVector{3, Float64}, 2)
2-element Vector{MVector{3, Float64}}:
 [0.0, 0.0, 0.0]
 [0.0, 0.0, 0.0]
```
Thus, I think something like
```julia
julia> MVector{3, Float64}(undef)
3-element MVector{3, Float64} with indices SOneTo(3):
 6.9325771233629e-310
 3.5e-323
 0.0
```
described in the docstring
```julia
help?> MVector
search: MVector @MVector

  MVector{S,T}(undef)
  MVector{S,T}(x::NTuple{S, T})
  MVector{S,T}(x1, x2, x3, ...)

  Construct a statically-sized, mutable vector MVector. Data may optionally be provided upon construction, and can
  be mutated later. Constructors may drop the T and S parameters if they are inferrable from the input (e.g.
  MVector(1,2,3) constructs an MVector{3, Int}).

  MVector{S}(vec::Vector)

  Construct a statically-sized, mutable vector of length S using the data from vec. The parameter S is mandatory
  since the length of vec is unknown to the compiler (the element type may optionally also be specified).
```
may be more clear to read.

Reply:
You are right. @amrueda, I recommend changing all occurrences of
```julia
f = zeros(MVector{nvariables(equations), eltype(u)})
```
to something like
```julia
f = MVector{nvariables(equations), eltype(u)}(undef)
f .= 0 # can be omitted if later code does not assume initialization to zero
```
for clarity. It is a little more verbose but also clearer to the user what's going on here.

Reply:
Just came to me: @ranocha what do you think about the alternative formulation
```julia
f = zero(MVector{nvariables(equations), eltype(u)})
```
In that case there should be no ambiguity of what is happening, and there is no extra docstring necessary (we *do* want the additive identity element for an `MVector`).

Reply:
That's fine with me as well :+1:

Reply:
Cool, thanks! Changed in 8662c8f